xapian-core/include/xapian/cluster.h

```diff
@@ -316,10 +407,30 @@ class XAPIAN_VISIBILITY_DEFAULT RoundRobin {
     ~RoundRobin();
 
     // Implements RoundRobin clustering
-    ClusterSet cluster(MSet &mset, unsigned int k);
+    ClusterSet cluster(MSet &mset, unsigned int num_of_clusters);
 
     // Returns the description of the clusterer being used
     std::string get_description();
 };
-}
+
+/** Kmeans clusterer:
+ *  This clusterer implements the K-Means clustering algorithm
+ */
+class XAPIAN_VISIBILITY_DEFAULT KMeans {
```

Comment:
I might have preferred to have the API refactor in one commit, and the KMeans implementation in another — it would have been easier to see what was going on with the refactor by only looking at the round robin version.